source/common/router/upstream_request.cc

```diff
@@ -478,6 +493,15 @@ void UpstreamRequest::enableDataFromDownstreamForFlowControl() {
   }
 }
 
+void UpstreamRequest::overflowDataFromDownstream() {
+  ASSERT(parent_.upstreamRequests().size() == 1 || parent_.downstreamEndStream());
+  // TODO(adip): Test counters
+  ENVOY_STREAM_LOG(warn, "Upstream request buffer overflow from single Downstream",
```

Comment:
Probably want to provide some control over whether these are warnings, as they might be noisy in situations where the OW limit is just not reasonably high enough yet.

Reply:
Should these be downgraded to `info`?
Does it make sense to keep them as warnings, and provide a user-configured threshold?

Reply:
If we could rate limit them, e.g. `LOG_EVERY_N_SECONDS`,  I think `warn` is fine, otherwise `debug`. Maybe with runtime control to switch.

Reply:
Adding `LOG_EVERY_N_SECONDS` or similar comes up at least once a month, and is not hard to implement. Let's do it. :)